In [61]:
import pandas as pd
df = pd.read_csv("anage_animalia.csv")
df

,HAGRID,Kingdom,Phylum,Class,Order,Family,Genus,Species,Common name,Female maturity (days),...,Source,Specimen origin,Sample size,Data quality,IMR (per yr),MRDT (yrs),Metabolic rate (W),Body mass (g),Temperature (K),References
0,32,Animalia,Chordata,Amphibia,Anura,Ascaphidae,Ascaphus,truei,Coastal tailed frog,NaN,...,525,wild,medium,acceptable,NaN,NaN,NaN,NaN,NaN,"525,749"
1,33,Animalia,Chordata,Amphibia,Anura,Bombinatoridae,Bombina,bombina,Firebelly toad,1095.0,...,485,captivity,medium,acceptable,NaN,NaN,NaN,NaN,NaN,"485,525,749"
2,34,Animalia,Chordata,Amphibia,Anura,Bombinatoridae,Bombina,orientalis,Oriental firebelly toad,NaN,...,451,captivity,large,acceptable,NaN,NaN,NaN,NaN,NaN,"451,525,749"
3,35,Animalia,Chordata,Amphibia,Anura,Bombinatoridae,Bombina,variegata,Yellow-bellied toad,NaN,...,NaN,captivity,medium,acceptable,NaN,NaN,NaN,NaN,NaN,"24,525,749"
4,36,Animalia,Chordata,Amphibia,Anura,Bufonidae,Anaxyrus,americanus,American toad,910.0,...,525,captivity,medium,acceptable,NaN,NaN,0.00829,21.6,288.0,"37,268,426,525,749"
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
4508,4595,Animalia,Chordata,Teleostei,Zeiformes,Oreosomatidae,Allocyttus,niger,Black oreo,9855.0,...,454,wild,medium,acceptable,NaN,NaN,NaN,NaN,NaN,454
4509,4596,Animalia,Chordata,Teleostei,Zeiformes,Oreosomatidae,Allocyttus,verrucosus,Warty oreo,NaN,...,22,wild,medium,acceptable,NaN,NaN,NaN,NaN,NaN,"22,454"
4510,4597,Animalia,Chordata,Teleostei,Zeiformes,Oreosomatidae,Neocyttus,rhomboidalis,Spiky oreo,NaN,...,454,wild,medium,acceptable,NaN,NaN,NaN,NaN,NaN,454
4511,4598,Animalia,Chordata,Teleostei,Zeiformes,Oreosomatidae,Pseudocyttus,maculatus,Smooth oreo,11315.0,...,454,wild,medium,acceptable,NaN,NaN,NaN,NaN,NaN,454


In [34]:
df.columns

Index(['HAGRID', 'Kingdom', 'Phylum', 'Class', 'Order', 'Family', 'Genus',
       'Species', 'Common name', 'Female maturity (days)',
       'Male maturity (days)', 'Gestation/Incubation (days)', 'Weaning (days)',
       'Litter/Clutch size', 'Litters/Clutches per year',
       'Inter-litter/Interbirth interval', 'Birth weight (g)',
       'Weaning weight (g)', 'Adult weight (g)', 'Growth rate (1/days)',
       'Maximum longevity (yrs)', 'Source', 'Specimen origin', 'Sample size',
       'Data quality', 'IMR (per yr)', 'MRDT (yrs)', 'Metabolic rate (W)',
       'Body mass (g)', 'Temperature (K)', 'References'],
      dtype='str')

In [35]:
df.isna().sum()

HAGRID                                 0
Kingdom                                0
Phylum                                 0
Class                                  0
Order                                  0
Family                                 0
Genus                                  0
Species                                0
Common name                            0
Female maturity (days)              2054
Male maturity (days)                2633
Gestation/Incubation (days)         2680
Weaning (days)                      3745
Litter/Clutch size                  2422
Litters/Clutches per year           3209
Inter-litter/Interbirth interval    3748
Birth weight (g)                    3283
Weaning weight (g)                  4129
Adult weight (g)                     872
Growth rate (1/days)                3954
Maximum longevity (yrs)              488
Source                               554
Specimen origin                        0
Sample size                            0
Data quality    

In [36]:
df[["Data quality", "Specimen origin", "Sample size"]]

,Data quality,Specimen origin,Sample size
0,acceptable,wild,medium
1,acceptable,captivity,medium
2,acceptable,captivity,large
3,acceptable,captivity,medium
4,acceptable,captivity,medium
...,...,...,...
4508,acceptable,wild,medium
4509,acceptable,wild,medium
4510,acceptable,wild,medium
4511,acceptable,wild,medium


In [37]:
df["Data quality"].value_counts()

Data quality
acceptable      3816
low              488
high             108
questionable     101
Name: count, dtype: int64

In [21]:
df["Specimen origin"].value_counts()

Specimen origin
wild         2008
captivity    1902
unknown       603
Name: count, dtype: int64

In [ ]:
df["Sample size"].value_counts()

Sample size
medium    2408
small     1404
large      382
tiny       318
huge         1
Name: count, dtype: int64

1)Al analizar la base de datos nos damos cuenta que existen demasiados datos inexistentes e varias columnas, en algunas estos datos inexistentes superan el 50% de la muestra, lo que sugiere que si analzamos estas columnas no sera un muy buen analisis

In [20]:
df.groupby("Class")["Species"].nunique().sort_values(ascending=False).reset_index()

,Class,Species
0,Aves,1202
1,Mammalia,1127
2,Teleostei,720
3,Reptilia,502
4,Amphibia,177
5,Chondrichthyes,115


2. La clase aves es la qeu tiene mayor numero de espcecies con 1202

In [59]:
faltantes = df.isna().sum()
porcentaje = (faltantes / len(df)) * 100
porcentaje
porcentaje.sort_values(ascending=False).head(5)

MRDT (yrs)              99.202304
IMR (per yr)            99.113672
Weaning weight (g)      91.491248
Temperature (K)         89.053844
Growth rate (1/days)    87.613561
dtype: float64

3.Estas anteriores son las que mas datos faltantes hay, y creo que es porque obtener este tipo de datos es mucho mas complicado que los otros, ya que son variables que camb ian con el tiempo.

In [62]:
DTF=df.query("`Data quality` == 'high' or `Data quality` == 'acceptable'")
DTF["Species"].nunique()

2940

In [63]:
df["Species"].nunique()

3290

In [57]:
2940/3290

0.8936170212765957

4.El 89% de las especies tiene high o aceptable en data quality sobreviviendo 2940 de 3290 

In [64]:
DTF["Maximum longevity (yrs)"].describe().reset_index()

,index,Maximum longevity (yrs)
0,count,3924.000000
1,mean,18.952232
2,std,17.092274
3,min,0.160000
4,25%,9.000000
5,50%,15.000000
6,75%,24.000000
7,max,392.000000


In [65]:
DTF.groupby("Class")["Maximum longevity (yrs)"].mean().sort_values(ascending = False).reset_index()

,Class,Maximum longevity (yrs)
0,Chondrichthyes,25.453448
1,Reptilia,22.057534
2,Mammalia,19.954437
3,Aves,18.016939
4,Teleostei,17.017960
5,Amphibia,15.371429


In [44]:
df.groupby("Specimen origin")["Maximum longevity (yrs)"].describe().reset_index()

,Specimen origin,count,mean,std,min,25%,50%,75%,max
0,captivity,1902.0,20.298107,14.035431,0.80,11.025,17.8,26.0,177.0
1,unknown,115.0,17.508696,13.461573,2.00,8.500,12.5,24.0,75.0
2,wild,2008.0,17.426524,19.455218,0.16,7.800,12.0,21.0,392.0


In [47]:
pf = df.sort_values("Maximum longevity (yrs)", ascending = True)
pf["Maximum longevity (yrs)"]

4090    0.16
1276    0.60
3906    0.80
3851    1.00
4089    1.00
        ... 
3853     NaN
3868     NaN
3870     NaN
4026     NaN
4504     NaN
Name: Maximum longevity (yrs), Length: 4513, dtype: float64